test/integration/trackedProgresses.js

```diff
@@ -0,0 +1,457 @@
+const { updateTask } = require("../../models/tasks");
+const {
+  predefinedTrackedProgressDataForUser,
+  predefinedTrackedProgressDataForTask,
+  isISOString,
+  trackedProgressUserDataForPost,
+  trackedProgressTaskDataForPost,
+  trackedProgressDataForPatch,
+} = require("../fixtures/trackedProgress");
+
+const userData = require("../fixtures/user/user")();
+const [userData0, userData1, , , superUserData] = userData;
+
+const cookieName = config.get("userToken.cookieName");
+const { expect } = chai;
+
+/* eslint-disable mocha/no-skipped-tests */
+describe.skip("Test the tracked Progress API", function () {
+  let userId0, userId1, superUser;
+  let userIdToken0, superUserToken;
+  let taskId0, taskId1;
+
+  beforeEach(async function () {
+    userId0 = await addUser(userData0);
+    userIdToken0 = authService.generateAuthToken({ userId: userId0 });
+    userId1 = await addUser(userData1);
+    superUser = await addUser(superUserData);
+    superUserToken = authService.generateAuthToken({ userId: superUser });
+
+    const taskObject0 = await updateTask(taskData[0]);
+    taskId0 = taskObject0.taskId;
+    const taskObject1 = await updateTask(taskData[1]);
+    taskId1 = taskObject1.taskId;
+
+    const docRefUser0 = firestore.collection("trackedProgresses").doc();
+    await docRefUser0.set({ ...predefinedTrackedProgressDataForUser, userId: userId0 });
+
+    const docRefTask0 = firestore.collection("trackedProgresses").doc();
+    await docRefTask0.set({ ...predefinedTrackedProgressDataForTask, taskId: taskId0 });
+  });
+
+  afterEach(async function () {
+    await cleanDb();
+  });
+
+  describe("Verify the POST call for tracked progresses", function () {
+    it("stores the tracked progress document for user", async function () {
+      const response = await chai
+        .request(app)
+        .post("/tracked-progresses")
+        .set("cookie", `${cookieName}=${superUserToken}`)
+        .send({
+          ...trackedProgressUserDataForPost,
+          userId: userId1,
+        });
+      expect(response).to.have.status(201);
+      expect(response.body).to.be.an("object");
+      expect(response.body).to.have.property("message").that.equals("Resource created successfully.");
+      expect(response.body).to.have.property("data").that.is.an("object");
+
+      const data = response.body.data;
+      expect(data).to.have.keys(["id", "type", "userId", "currentlyTracked", "frequency", "createdAt", "updatedAt"]);
+      expect(data).to.have.property("id").that.is.a("string");
+      expect(data).to.have.property("type").that.equals("user");
+      expect(data).to.have.property("userId").that.equals(userId1);
+      expect(data).to.have.property("currentlyTracked").that.equals(true);
+      expect(data).to.have.property("frequency").that.equals(1);
+      expect(data).to.have.property("createdAt").that.is.a("string").and.satisfy(isISOString);
+      expect(data).to.have.property("updatedAt").that.is.a("string").and.satisfy(isISOString);
```

Comment:
This feels too verbose.
Since we already checked for keys we can write other statements like this:
```js
expect(data.id).to.be.equal("123")
```